src/main/webapp/app/lecture/lecture-update.component.ts

```diff
@@ -38,10 +38,18 @@ export class LectureUpdateComponent implements OnInit {
      */
     ngOnInit() {
         this.isSaving = false;
-        // Wait for both subscriptions to load
-        combineLatest(this.activatedRoute.data, this.activatedRoute.parent!.data).subscribe(([routeData, parentRouteData]) => {
-            this.lecture = routeData['lecture'];
-            this.lecture.course = parentRouteData['course'];
+        // Create a new lecture to use unless we fetch an existing lecture
+        this.lecture = new Lecture();
+        this.activatedRoute.parent!.data.subscribe((data) => {
+            const lecture = data['lecture'];
+            if (lecture) {
```

Comment:
You could shorten line 42 and 44-47 to `this.lecture = data['lecture'] ? data['lecture'] : new Lecture()`. Or also `this.lecture = lecture ? lecture : new Lecture()` if you want to keep the lecture constant. would save assigning this.lecture two times

Reply:
I changed it to `this.lecture = lecture ?? new Lecture();` which is even more concise.